invenio_app_rdm/administration/records/records.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+# Copyright (C) 2023 Graz University of Technology.
+#
+# invenio-administration is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Invenio administration OAI-PMH view module."""
+from functools import partial
+
+from flask import current_app
+from invenio_administration.views.base import AdminResourceListView
+from invenio_i18n import lazy_gettext as _
+from invenio_search_ui.searchconfig import search_app_config
+
+
+class RecordAdminListView(AdminResourceListView):
+    """Configuration for OAI-PMH sets list view."""
+
+    api_endpoint = "/records/all"
+    name = "Records"
+    resource_config = "records_resource"
+    search_request_headers = {"Accept": "application/json"}
+    title = "Records"
+    category = "Records & files"
+    icon = "file"
+    template = "invenio_app_rdm/administration/records/record-admin-search.html"
```

Comment:
I would make it shorter, or `records-search.html` (for consistency with `users_search.html`), we are already in the `administration` module:
```suggestion
    template = "invenio_app_rdm/administration/records/records.html"
```